coordinator/src/main/scala/filodb.coordinator/queryplanner/LongTimeRangePlanner.scala

```diff
@@ -90,12 +93,21 @@ class LongTimeRangePlanner(rawClusterPlanner: QueryPlanner,
 
     val onKeysReal = ExtraOnByKeysUtil.getRealOnLabels(logicalPlan, queryConfig.addExtraOnByKeysTimeRanges)
 
+
+    val dispatcher = if (lhsExec.dispatcher.isInstanceOf[ActorPlanDispatcher] &&
+      rhsExec.dispatcher.isInstanceOf[ActorPlanDispatcher]) {
+      val lhsCluster = lhsExec.dispatcher.asInstanceOf[ActorPlanDispatcher].clusterName
+      val rhsCluster = rhsExec.dispatcher.asInstanceOf[ActorPlanDispatcher].clusterName
+      if (rhsCluster.equals(lhsCluster)) PlannerUtil.pickDispatcher(lhsExec.children ++ rhsExec.children)
+      else inProcessPlanDispatcher
+    } else inProcessPlanDispatcher
```

Comment:
`clusterName`  should be in PlanDispatcher trait. This will avoid the type casts.

Reply:
Added clusterName to PlanDispatcher